src/main/scala/com.snowplowanalytics.sauna/responders/hipchat/SendRoomNotificationResponder.scala

```diff
@@ -0,0 +1,238 @@
+              case None =>
+                Some(RoomNotificationReceived(data, observerEvent))
+              case Some(error) =>
+                logger ! Notification(error)
+                None
+            }
+          case Left(error) =>
+            logger ! Notification(error)
+            None
+        }
+      case None =>
+        logger ! Notification("No stream present, cannot parse command")
+        None
+    }
+  }
+
+  /**
+   * Send a valid room notification using the API wrapper.
+   *
+   * @param event The event containing a room notification.
+   */
+  override def process(event: RoomNotificationReceived): Unit =
+    hipchat.sendRoomNotification(event.roomNotification).onComplete {
+      case Success(message) => context.parent ! RoomNotificationSent(event, s"Successfully sent HipChat notification: $message")
+      case Failure(error) => logger ! Notification(s"Error while sending HipChat notification: $error")
+    }
+}
+
+object SendRoomNotificationResponder {
+  /**
+   * A responder event denoting that a HipChat room notification was received by
+   * the responder.
+   *
+   * @param roomNotification A room notification represented as a case class.
+   * @param source           The observer event that triggered this responder event.
+   */
+  case class RoomNotificationReceived(
+    roomNotification: RoomNotification,
+    source: ObserverBatchEvent) extends ResponderEvent[ObserverBatchEvent]
+
+  /**
+   * A responder result denoting that a HipChat room notification was successfully sent
+   * by the responder.
+   *
+   * @param source  The responder event that triggered this.
+   * @param message A success message.
+   */
+  case class RoomNotificationSent(
+    source: RoomNotificationReceived,
+    message: String) extends ResponderResult
+
+  /**
+   * A self-describing JSON structure, containing a reference to a JSON
+   * schema and data that should validate against the given schema.
+   *
+   * @param schema A self-describing schema key.
+   * @param data   Data that should comply to the schema.
+   */
+  case class SelfDescribing(
+    schema: SchemaKey,
+    data: JsValue)
+
+  implicit val selfDescribingReads: Reads[SelfDescribing] = (
+    (JsPath \ "schema").read[String].map[SchemaKey](schemaKey => SchemaKey.fromUri(schemaKey).get) and
+      (JsPath \ "data").read[JsValue]
+    ) (SelfDescribing.apply _)
+
+  /**
+   * A Sauna command.
+   *
+   * @param envelope Identifies the command and provides instructions on how to execute it.
+   * @param command  Specifies the exact action to execute.
+   */
+  case class SaunaCommand(
+    envelope: SelfDescribing,
+    command: SelfDescribing)
+
+  implicit val saunaCommandReads: Reads[SaunaCommand] = (
+    (JsPath \ "envelope").read[SelfDescribing] and
+      (JsPath \ "command").read[SelfDescribing]
+    ) (SaunaCommand.apply _)
+
+  /**
+   * TODO: define what this is
+   */
+  object Semantics extends Enumeration {
```

Comment:
[Sealed hierarchy](http://underscore.io/blog/posts/2015/06/02/everything-about-sealed.html) with only case objects is more Scala-ish way to implement enumerables.

```scala
sealed trait Semantics
case object AtLeastOneSemantics extends Semantics
// etc
```